src/wp-includes/rest-api/endpoints/class-wp-rest-block-directory-controller.php

```diff
@@ -0,0 +1,514 @@
+		}
+
+		require_once ABSPATH . 'wp-admin/includes/plugin-install.php';
+		require_once ABSPATH . 'wp-admin/includes/plugin.php';
+
+		$response = plugins_api(
+			'query_plugins',
+			array(
+				'block'    => $search_string,
+				'per_page' => 3,
+			)
+		);
+
+		if ( is_wp_error( $response ) ) {
+			return $response;
+		}
+
+		$result = array();
+
+		foreach ( $response->plugins as $plugin ) {
+			$installed_plugins = get_plugins( '/' . $plugin['slug'] );
+
+			// Only show uninstalled blocks.
+			if ( empty( $installed_plugins ) ) {
+				$data = $this->prepare_item_for_response( $plugin, $request );
+				$result[] = $this->prepare_response_for_collection( $data );
+			}
+		}
+
+		return rest_ensure_response( $result );
+	}
+
+	/**
+	 * Determine if the endpoints are available.
+	 *
+	 * Only the 'Direct' filesystem transport, and SSH/FTP when credentials are stored are supported at present.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @return bool|WP_Error True if filesystem is available, WP_Error otherwise.
+	 */
+	private static function is_filesystem_available() {
+		$filesystem_method = get_filesystem_method();
+
+		if ( 'direct' === $filesystem_method ) {
+			return true;
+		}
+
+		ob_start();
+		$filesystem_credentials_are_stored = request_filesystem_credentials( self_admin_url() );
+		ob_end_clean();
+
+		if ( $filesystem_credentials_are_stored ) {
+			return true;
+		}
+
+		return new WP_Error( 'fs_unavailable', __( 'The filesystem is currently unavailable for installing blocks.' ) );
+	}
+
+	/**
+	 * Parse block metadata for a block, and prepare it for an API repsonse.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param WP_Object $plugin The plugin metadata.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_Error|WP_REST_Response Response object on success, or WP_Error object on failure.
+	 */
+	public function prepare_item_for_response( $plugin, $request ) {
+
+		// There might be multiple blocks in a plugin. Only the first block is mapped.
+		$block_data   = reset( $plugin['blocks'] );
+
+		// A data array containing the properties we'll return.
+		$block = array(
+			'name'                  => $block_data['name'],
+			'title'                 => ( $block_data['title'] ? $block_data['title'] : $plugin['name'] ),
+			'description'           => wp_trim_words( $plugin['description'], 30, '...' ),
+			'id'                    => $plugin['slug'],
+			'rating'                => $plugin['rating'] / 20,
+			'rating_count'          => intval( $plugin['num_ratings'] ),
+			'active_installs'       => intval( $plugin['active_installs'] ),
+			'author_block_rating'   => $plugin['author_block_rating'] / 20,
+			'author_block_count'    => intval( $plugin['author_block_count'] ),
+			'author'                => wp_strip_all_tags( $plugin['author'] ),
+			'icon'                  => ( isset( $plugin['icons']['1x'] ) ? $plugin['icons']['1x'] : 'block-default' ),
+			'assets'                => array(),
+			'humanized_updated'     => sprintf(
```

Comment:
Just returning the `last_updated` would be closer to how the REST API generally tries to expose data. Ie exposing the underlying value instead of formatted for a specific context. But I think Gutenberg still doesn't have a `human_time_diff` equivalent. So if we need to keep it, I'd just also expose the raw `last_updated` property.